configuration/rules-blockly-run-and-process.md

```diff
@@ -0,0 +1,163 @@
+
+### Retrieve rule context information
+
+![retrieve-context-info](images/blockly/blockly-retrieve-rule-context-info.png)
+
+*Function:* Retrieve event context related information of the rule
+
+A rule carries contextual information when triggered - this block can retrieve that information.
+
+* The rule trigger type defines which of the attributes are available to the rule - they will not all be available.
+* For more detailed information see [Event Object Attributes ](https://openhab-scripters.github.io/openhab-helper-libraries/Guides/Event%20Object%20Attributes.html)
+
+## Transform values via Map, Regex or JsonPath and others
+
+OpenHAB provides many [transformations](https://www.openhab.org/docs/configuration/transformations.html):
+- [Map](https://www.openhab.org/addons/transformations/map/)
+- [Regex](https://www.openhab.org/addons/transformations/regex/)
+- [JsonPath](https://www.openhab.org/addons/transformations/jsonpath/)
+- [Binary to Json](https://www.openhab.org/addons/transformations/bin2json/)
+- [Exec](https://www.openhab.org/addons/transformations/exec/)
+- [JavaScript](https://www.openhab.org/addons/transformations/javascript/)
+- [Jinja](https://www.openhab.org/addons/transformations/jinja/)
+- [Scale](https://www.openhab.org/addons/transformations/scale/)
+- [XPath](https://www.openhab.org/addons/transformations/xpath/)
+- [XSLT](https://www.openhab.org/addons/transformations/xslt/)
+
+The following describes examples for Map, Regex, and JsonPath
+
+![map](images/blockly/blockly-transform-map.png)
+![regex](images/blockly/blockly-transform-regex.png)
+![jsonpath](images/blockly/blockly-transform-jsonpath.png)
+
+Function: Transform the given value via one of the different transformation methods
+
+### Example 1: Regex
+
+Convert  #123456 into rgb(12,34,56)
+
+![transform-regex](images/blockly/blockly-transform-example1.png)
+
+### Example 2: JsonPath transformation
+Extract the temperature from the following JSON
+```json
+{ "device": { "location": "Outside", "status": { "temperature": 23.2 }}}
+```
+by using the jsonpath
+
+```json
+$.device.status.temperature
+```
+
+![transform-jsonpath](images/blockly/blockly-transform-example2.png)
+
+### Example 3: Map transformation
+
+This example applies a map transformation from the map file nanoleaf.map in *$OPENHAB-CONF/transform/*
+
+![transform-map](images/blockly/blockly-transform-example3.png)
+
+The above, with the below map file
+
+```text
+ON=Ja
+OFF=Nein
+effects=Effekte
+-=?
+NULL=n/a
+hs = Hue/Saturation
+ct = Color Temperature
+```
+
+would result in the following log entry
+
+```bash
+Hue/Saturation
+```
+
+## Inline Script
+
+![inline-script](images/blockly/blockly-inline-script.png)
+
+*Function:* execute arbitrary ECMAScript content
+
+Provides the possibility to inject any ECMAScript at a given position. This should be used in cases when Blockly does not provide the desired functionality. Use with diligence.
```

Comment:
Please change this to one sentence per line.
The website build will take core of a proper presentation of paragraphs.